cg/meta/orders/ticket_handler.py

```diff
@@ -45,22 +46,33 @@ def create_ticket(
             order=order,
             project=project,
         )
-        attachment: dict = self.create_attachment(order=order)
-        ticket_nr: str | None = self.osticket.open_ticket(
-            name=user_name,
-            email=user_mail,
-            subject=order.name,
-            message=message,
-            attachment=attachment,
-        )
-        LOG.info(f"{ticket_nr}: opened new ticket")
 
-        return ticket_nr
+        with TemporaryDirectory() as temp_dir:
+            attachments: List[Path] = self.create_attachments(order=order, temp_dir=temp_dir)
 
-    def create_attachment(self, order: OrderIn):
-        return self.osticket.create_new_ticket_attachment(
-            content=self.replace_empty_string_with_none(obj=order.dict()), file_name="order.json"
-        )
+            # Create ticket in Freshdesk
+            freshdesk_ticket = TicketCreate(
+                email=user_mail,
+                description=message,
+                name=user_name,
+                subject=order.name,
+                attachments=[],
+            )
+            ticket_response: TicketResponse = self.client.create_ticket(
+                ticket=freshdesk_ticket, attachments=attachments
+            )
+            LOG.info(f"{ticket_response.id}: opened new ticket in Freshdesk")
+
+        return ticket_response.id
+
+    def create_attachments(self, order: OrderIn, temp_dir: str) -> List[Path]:
+        """Create attachments for the ticket"""
+        attachments = []
+        order_file_path = Path(temp_dir) / "order.json"
+        with order_file_path.open("w") as order_file:
+            order_file.write(order.json())
+        attachments.append(order_file_path)
+        return attachments
```

Comment:
Is there a reason we create a list here? I think you can simplify this a bit
```python
def create_attachment(self, order: OrderIn, temp_dir: str) -> Path:
    """Create attachments for the ticket."""
    order_file_path = Path(temp_dir) / "order.json"
    order_file_path.write_text(order.json())
    return order_file_path
```

Reply:
What I understood after a quick look at Freshdesk ticket creation [API](https://developers.freshdesk.com/api/#create_ticket_with_attachment) is that it handles attachments as a list. Here is an [example](https://github.com/freshworks/fresh-samples/blob/master/Python/create_ticket_attachment.py)

Reply:
Alright, then wrap it in a list when you send the request! 

The code becomes clearer if you shift that logic (that it needs to be a list) to where the request is sent against fresh desk.

Reply:
Thanks ! I think it is clear this way ..